R/build_copy_queries.R

```diff
@@ -220,6 +223,12 @@ build_copy_queries <- function(dest, dm, set_key_constraints = TRUE, temporary =
 
   queries <- left_join(create_table_queries, index_queries, by = "name")
 
+  queries$dml <- dm_dml_load(
+    dm, con,
+    ## queries DF order does not match to table_names order so we need to match to order here
+    table_names = table_names[match(queries$name, names(table_names))]
+  )
+
   ## Reorder queries according to topological sort so pks are created before associated fks
   graph <- create_graph_from_dm(dm, directed = TRUE)
   topo <- igraph::topo_sort(graph, mode = "in")
```

Comment:
This code and below computes the correct creation and insertion order. Perhaps this part should move further up in the function, we could reorder the tables in the dm object according to that order.

Reply:
This is not relevant because queries are not executed at that point, so reordering here is fine.